### PR TITLE
Move `RadioControl`and `RadioControlAccordion` components to components package

### PR DESCRIPTION
**Pull request by alexflorisca**
## What
This PR is based on https://github.com/woocommerce/woocommerce-blocks/pull/11214 (update/checkbox-component)
Move `RadioControl` and `RadioControlAccordion` components to components package and update the places it is imported:

- ShippingRates
- PaymentMethods
- PickupOptions
- LocalPickup

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Install [Multiple Packages for WooCommerce](https://wordpress.org/plugins/multiple-packages-for-woocommerce/) and go to WooCommerce > Settings > Multiple Packages. Set the "Group by" option to Product (individual).
2. Create at least 2 pickup locations (WooCommerce > Settings > Shipping > Local Pickup)
3. Add multiple products to your cart. Ensure you add one product more than once (so you have 2+ of a single item in the cart)
4. Visit the checkout
5. In the shipping options section, switch between different options and make sure the right option is selected and they are greyed out while the http request is in progress.
6. In the shipping method section, switch to Local Pickup
7. Select one of the other pickup locations and make sure the right option is selected.
8. Scroll down to the Payment Options section.
9. Change the payment method and make sure the right option is selected.
10. Fill in all the other details, place the order and make sure it goes through ok.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Make the `RadioControl` and `RadioControlAccordion` component available in the @woocommerce/blocks-components package.

### CODE REVIEW

File: bin/webpack-entries.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ const getBlockEntries = ( relativePath ) => {
 const entries = {
 	styling: {
 		// Packages styles
-		'packages-style': glob.sync( './packages/**/index.js' ),
+		'packages-style': glob.sync( './packages/**/index.{t,j}s' ),
```
Comment (opr) [inline bin/webpack-entries.js:169]: Nice spot!

**Review by opr — CHANGES_REQUESTED** [review]
Hey, so the changes look good and I know you struggled with the types yesterday, however what do you think of this? I have provided the patch file for my changes, but I can just go ahead and commit to your branch, let me know!


```patch
diff --git a/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx b/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx
index 2404e4f5b..696db0fef 100644
--- a/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx
+++ b/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx
@@ -1,12 +1,11 @@
 /**
  * External dependencies
  */
-import { RadioControl } from '@woocommerce/blocks-components';
+import {
+	RadioControl,
+	RadioControlOptionType,
+} from '@woocommerce/blocks-components';
 import { CartShippingPackageShippingRate } from '@woocommerce/types';
-/**
- * Internal dependencies
- */
-import type { RadioControlOption } from '../../../../../../packages/components/radio-control/types';
 
 interface LocalPickupSelectProps {
 	title?: string | undefined;
@@ -17,7 +16,7 @@ interface LocalPickupSelectProps {
 	renderPickupLocation: (
 		location: CartShippingPackageShippingRate,
 		pickupLocationsCount: number
-	) => RadioControlOption;
+	) => RadioControlOptionType;
 	packageCount: number;
 }
 /**
diff --git a/assets/js/blocks/cart-checkout-shared/payment-methods/saved-payment-method-options.tsx b/assets/js/blocks/cart-checkout-shared/payment-methods/saved-payment-method-options.tsx
index a50974a2e..c6a11c25a 100644
--- a/assets/js/blocks/cart-checkout-shared/payment-methods/saved-payment-method-options.tsx
+++ b/assets/js/blocks/cart-checkout-shared/payment-methods/saved-payment-method-options.tsx
@@ -4,7 +4,10 @@
 import { useMemo, cloneElement } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 import { noticeContexts } from '@woocommerce/base-context';
-import { RadioControl } from '@woocommerce/blocks-components';
+import {
+	RadioControl,
+	RadioControlOptionType,
+} from '@woocommerce/blocks-components';
 import {
 	usePaymentMethodInterface,
 	useStoreEvents,
@@ -19,7 +22,6 @@ import { isNull } from '@woocommerce/types';
  */
 import { getCanMakePaymentArg } from '../../../data/payment/utils/check-payment-methods';
 import { CustomerPaymentMethodConfiguration } from '../../../data/payment/types';
-import type { RadioControlOption } from '../../../../../packages/components/radio-control/types';
 
 /**
  * Returns the option object for a cc or echeck saved payment method token.
@@ -87,7 +89,7 @@ const SavedPaymentMethodOptions = () => {
 	const { removeNotice } = useDispatch( 'core/notices' );
 	const { dispatchCheckoutEvent } = useStoreEvents();
 
-	const options = useMemo< RadioControlOption[] >( () => {
+	const options = useMemo< RadioControlOptionType[] >( () => {
 		const types = Object.keys( savedPaymentMethods );
 
 		// Get individual payment methods from saved payment methods and put them into a unique array.
@@ -145,7 +147,7 @@ const SavedPaymentMethodOptions = () => {
 		} );
 		return mappedOptions.filter(
 			( option ) => typeof option !== 'undefined'
-		) as RadioControlOption[];
+		) as RadioControlOptionType[];
 	}, [
 		savedPaymentMethods,
 		paymentMethods,
diff --git a/assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/block.tsx b/assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/block.tsx
index 454a4b4b7..68b3f8a01 100644
--- a/assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/block.tsx
+++ b/assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/block.tsx
@@ -10,7 +10,10 @@ import {
 } from '@wordpress/element';
 import { useShippingData, useStoreCart } from '@woocommerce/base-context/hooks';
 import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
-import { FormattedMonetaryAmount } from '@woocommerce/blocks-components';
+import {
+	FormattedMonetaryAmount,
+	RadioControlOptionType,
+} from '@woocommerce/blocks-components';
 import { decodeEntities } from '@wordpress/html-entities';
 import { getSetting } from '@woocommerce/settings';
 import { Icon, mapMarker } from '@wordpress/icons';
@@ -26,7 +29,6 @@ import { LocalPickupSelect } from '@woocommerce/base-components/cart-checkout/lo
  * Internal dependencies
  */
 import ShippingRatesControlPackage from '../../../../base/components/cart-checkout/shipping-rates-control-package';
-import type { RadioControlOption } from '../../../../../../packages/components/radio-control/types';
 
 const getPickupLocation = (
 	option: CartShippingPackageShippingRate
@@ -67,7 +69,7 @@ const getPickupDetails = (
 const renderPickupLocation = (
 	option: CartShippingPackageShippingRate,
 	packageCount: number
-): RadioControlOption => {
+): RadioControlOptionType => {
 	const priceWithTaxes = getSetting( 'displayCartPricesIncludingTax', false )
 		? parseInt( option.price, 10 ) + parseInt( option.taxes, 10 )
 		: option.price;
diff --git a/packages/components/index.ts b/packages/components/index.ts
index 255e6d1f6..83a10c64f 100644
--- a/packages/components/index.ts
+++ b/packages/components/index.ts
@@ -4,6 +4,7 @@ export {
 	RadioControlOption,
 	RadioControlOptionLayout,
 } from './radio-control';
+export type { RadioControlOption as RadioControlOptionType } from './radio-control/types';
 export { default as FormattedMonetaryAmount } from './formatted-monetary-amount';
 export { Chip, RemovableChip } from './chip';
 export { default as RadioControlAccordion } from './radio-control-accordion';
```

**Comment by alexflorisca** [comment]
Thanks for the review @opr. I can't quite figure out from that diff what changes you made from what we had before? We seem to still be exporting type which didn't work before?

**Comment by opr** [comment]
> Thanks for the review @opr. I can't quite figure out from that diff what changes you made from what we had before? We seem to still be exporting type which didn't work before?

@alexflorisca I can't remember exactly what we had before, but I think the main difference is that I renamed the type to `RadioControlOptionType` so exporting it works better, I think there was a type _and_ a component with the same name before?

**Comment by alexflorisca** [comment]
I definitely renamed it to `RadioControlOptionType` as well, but I think the difference might be I didn't `export type` only `import type`. So that line wouldn't have been stripped out by the ts compiler. Feel free to commit here if you already have a working version.

`export type { RadioControlOption as RadioControlOptionType } from './radio-control/types';`

**Comment by opr** [comment]
@alexflorisca committed in https://github.com/woocommerce/woocommerce-blocks/pull/11312/commits/2c93a407be56c666230a992fab6bf3404dbe9192

**Comment by alexflorisca** [comment]
Nice one. I've rebased trunk, once the CI passes it should be good to go, just need a 👍 from you.

**Review by opr — APPROVED** [review]

**Pull request merged**